src/components/ImageUpload.tsx

```diff
@@ -1,65 +1,98 @@
-import { Paper, Typography } from "@mui/material";
-import { styled } from "@mui/material/styles";
-import { Box } from "@mui/system";
-import { Field, FormikErrors, FormikTouched } from "formik";
-import React from "react";
-import { render } from "react-dom";
+import { Box, Typography } from "@mui/material";
+import { Image } from "cloudinary-react";
+import  { useCallback, useState } from "react";
 import { useDropzone } from "react-dropzone";
-import { theme } from "./../theme/theme";
-import { StyledPaperInput } from "./StyledComponents";
+import { StyledFilledButton, StyledPaperInput } from "./StyledComponents";
 
-interface ImageUploadProps {
-  setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
-  errors: FormikErrors<{
-    description: string;
-    genre: string;
-    imageFile: {
-      name: string;
-      size: string;
-      type: string;
-    };
-  }>;
-  touched: FormikTouched<{
-    description: string;
-    genre: string;
-    imageFile: {
-      name: string;
-      size: null;
-      type: string;
-    };
-    releaseDate: string;
-    title: string;
-    yourRole: string;
-  }>;
-}
+// interface ImageUploadProps {
+//   setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
+//   errors: FormikErrors<{
+//     description: string;
+//     genre: string;
+//     imageFile: {
+//       name: string;
+//       size: string;
+//       type: string;
+//     };
+//   }>;
+//   touched: FormikTouched<{
+//     description: string;
+//     genre: string;
+//     imageFile: {
+//       name: string;
+//       size: null;
+//       type: string;
+//     };
+//     releaseDate: string;
+//     title: string;
+//     yourRole: string;
+//   }>;
+// }
+
+export const ImageUpload = () => {
+  const PUBLIC_CLOUDINARY_CLOUD_NAME = "projectnewm",
+    PUBLIC_CLOUDINARY_UPLOAD_PRESET = "rvktckuk";
+
+  const [uploadedFilePublicId, setUploadedFilePublicId] = useState<string>();
+  const onDrop = useCallback(acceptedFiles => {
+    const url = `https://api.cloudinary.com/v1_1/${PUBLIC_CLOUDINARY_CLOUD_NAME}/upload`;
+
+    acceptedFiles.forEach(async (acceptedFile: File) => {
+      const formData = new FormData();
+      formData.append("file", acceptedFile);
+      formData.append("upload_preset", PUBLIC_CLOUDINARY_UPLOAD_PRESET);
+
+      const response = await fetch(url, {
```

Comment:
Just so I understand it better, is this an unauthenticated endpoint to upload images?
So anyone with the URL can attempt to upload files? If so, are we able to at least restrict it to a domain whitelist?

Reply:
This is an unsigned upload endpoint. I can implement a signed endpoint that should restrict who can upload files.